client/my-sites/themes/theme-upload/index.jsx

```diff
@@ -139,7 +139,11 @@ class Upload extends React.Component {
 
 		// DropZone supplies an array, FilePicker supplies a FileList
 		const file = files[ 0 ] || files.item( 0 );
-		if ( file.type !== 'application/zip' ) {
+
+		const validFileType = file.type === 'application/zip' || file.type === 'application/x-zip-compressed';
```

Comment:
I always liked this style of writing code.
First it removes decision logic from `if` and makes `if` line act only as switch.
Second it gives name and thus explains what the test is about.
Actually i think that I should even call it `isValidFileType` to make it better. 
This is more obvious whay to go when the test is more complex( maybe we add 3rd mime )
The other thing I like to do is even for such a small tasks is to move the code to a separate function that in this case would be called and executed `isValidFileType( file.type )`.
I know that this is something very personal for every person that works with code :) So if you think  that this is not a good thing for calypso I don't mind I can change it. 
Long answer but I did not wanted to leave it as: _I always liked this style of writing code._ :)